src/main/scala/edu/knowitall/collection/immutable/graph/Graph.scala

```diff
@@ -72,8 +72,12 @@ class Graph[T](
   def connected(v: T, pred: DirectedEdge[T] => Boolean): Set[T] = {
     def rec(vertices: Set[T], last: Set[T]): Set[T] = {
       val neighbors: Set[T] = last.flatMap(this.neighbors(_, pred))
-      if (neighbors.isEmpty) vertices
-      else rec(vertices ++ neighbors, neighbors -- vertices)
+      if (neighbors.isEmpty) {
+        vertices
+      }
```

Comment:
In the future, feel free to cuddle the `else`; that's what we usually do, and it saves vertical space:
```scala
if (neighbors.isEmpty) {
  vertices
} else {
  rec(vertices ++ neighbors, neighbors -- vertices)
}
```